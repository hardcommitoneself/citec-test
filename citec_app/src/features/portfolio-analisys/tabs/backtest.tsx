// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Button } from '@mui/material'
import { DailyReturns } from 'features/backtest/components'
import { TotalBacktest } from 'features/portfolio-analisys/totals'
import { useAppNavigate } from 'features/utils/hooks'
import { BacktestRes } from 'features/utils/interfaces/backtest-api-resp'

interface BacktestProps {
   backtest: BacktestRes
}

export const Backtest = ({ backtest }: BacktestProps) => {
   const navigate = useAppNavigate()

   return (
      <div className="summary_content_wrapper">
         <div className="action_portfolio_analysis">
            <h1>Performance Backtest</h1>
            <div className="button_rebalance_positions">
               <Button
                  className="w-36 h-10"
                  variant="optimize"
                  onClick={() => navigate('/investment')}
               >
                  <img
                     src={require('assets/img/investment_white.png')}
                     style={{ marginRight: '15px' }}
                  />
                  <span className="text-base">Rebalance</span>
               </Button>
            </div>
         </div>

         <TotalBacktest backtest={backtest} />

         {backtest && (
            <>
               <div className="BacktestChartWrapper">
                  <div className="top_area">
                     <h1>Returns %</h1>
                  </div>
                  <div className="line_chart_wrapper_backtest">
                     <DailyReturns ts_backtest={backtest.ts_backtest} />
                  </div>
               </div>
            </>
         )}
      </div>
   )
}
