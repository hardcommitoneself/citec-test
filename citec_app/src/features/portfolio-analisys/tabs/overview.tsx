// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Button } from '@mui/material'
import { Icon } from 'assets/icons'
import { renderPopup } from 'features/popups/overview-popups'
import { useAppNavigate } from 'features/utils/hooks'
import { BacktestRes } from 'features/utils/interfaces/backtest-api-resp'
import { numberToFixed } from 'features/utils/to-fixed-number'
import { useAppSelector } from 'store/hooks'
import {
   Factors,
   Sectors,
   TrackingError,
   SegmentMatrix,
   KeyMetricsTable,
} from '../components'
import { TotalOverview } from '../totals/totals-overview'

interface OverviewTabProps {
   backtest: BacktestRes
}

export const OverviewTab = ({ backtest }: OverviewTabProps) => {
   const navigate = useAppNavigate()

   const portfolio_analysis = useAppSelector((state) => state.portfolio.analysis)

   const { tracking_expected } = backtest

   return (
      <div className="summary_content_wrapper">
         <div className="action_portfolio_analysis">
            <h1>Overview</h1>
            <div className="button_rebalance_positions">
               <Button
                  className="w-36 h-10"
                  variant="optimize"
                  onClick={() => navigate('/investment')}
               >
                  <img
                     src={require('assets/img/investment_white.png')}
                     style={{ marginRight: '15px' }}
                  />
                  <span className="text-base">Rebalance</span>
               </Button>
            </div>
         </div>

         <TotalOverview />

         <div className="RiskReturnGrid">
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Sector composition
                     <button
                        className="mx-2"
                        onClick={() => renderPopup('sectorCompositionAlert')}
                        data-testid="sector-test"
                     >
                        <Icon.Info />
                     </button>
                  </h1>
               </div>

               <div className="chartRiskReturn">
                  <Sectors sectors={portfolio_analysis?.sectors as never} />
               </div>
            </div>

            <KeyMetricsTable backtest={backtest} />

            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Segment matrix
                     <button
                        className="mx-2"
                        onClick={() => renderPopup('segmentMatrixAlert')}
                     >
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="chartRiskReturn">
                  <SegmentMatrix
                     segmentmatrix={portfolio_analysis!.segment_matrix}
                  />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Factors
                     <button
                        className="mx-2"
                        onClick={() => renderPopup('factorsAlert')}
                     >
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="chartRiskReturn">
                  <Factors factors={portfolio_analysis?.factors} />
               </div>
            </div>
         </div>

         <div className="BacktestChartWrapper">
            <div className="rowBack">
               <p className="!text-[#2D405A] text-[18px] font-[600]">
                  Tracking Error probability distribution
               </p>

               <button
                  className="mx-2"
                  onClick={() => renderPopup('trackingErrorAlert')}
               >
                  <Icon.Info />
               </button>
            </div>
            <div className="rowBack">
               <h1 className="text-[20px]">Tracking Error: </h1>
               <div className="tracking_error_green">
                  {numberToFixed(
                     tracking_expected.expected_tracking_kpis.tracking_error * 100
                  )}
                  %
               </div>
            </div>
            <TrackingError tracking_expected={backtest.tracking_expected} />
         </div>
      </div>
   )
}
