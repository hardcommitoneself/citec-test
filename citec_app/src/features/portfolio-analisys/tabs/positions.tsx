// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { Button } from '@mui/material'
import { Table } from 'features/ui/table'
import { columns } from 'features/ui/table/columns/analysis-positions'
import { useAppNavigate } from 'features/utils/hooks'
import { useAppSelector } from 'store/hooks'
import { TotalPositions } from '../totals'

export const Positions = ({ backtest }: any) => {
   const navigate = useAppNavigate()

   const positions = useAppSelector((state) => state.portfolio.positions)

   return (
      <div className="summary_content_wrapper">
         <div className="action_portfolio_analysis">
            <h1>Positions</h1>
            <div className="button_rebalance_positions">
               <Button
                  className="w-36 h-10"
                  variant="optimize"
                  onClick={() => navigate('/investment')}
               >
                  <img
                     src={require('assets/img/investment_white.png')}
                     style={{ marginRight: '15px' }}
                  />
                  <span className="text-base">Rebalance</span>
               </Button>
            </div>
         </div>

         <TotalPositions backtest={backtest} />

         <Table columns={columns} data={positions as never} />
      </div>
   )
}
