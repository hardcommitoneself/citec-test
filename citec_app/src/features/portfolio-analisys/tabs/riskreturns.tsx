// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import Swal from 'sweetalert2'
import { Concentration, RiskReturns, ValueAtRisk } from '../components'
import { formatToMoney } from 'features/utils/format-to-money'
import { Icon } from 'assets/icons'
import { useAppSelector } from 'store/hooks'

export const RiskreturnTab = () => {
   const portfolio_analysis = useAppSelector((state) => state.portfolio.analysis)

   const valueAtRiskAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Value at risk</b> quantifies the portfolio’s possible financial losses 
         (in dollars or in percentage) over a specific time frame at a 
         defined confidence level..
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const riskVsReturnAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Risk vs. return</b> analysis is used to determine how close your portfolio is to 
         the efficiency frontier, compared to the benchmark. The efficiency frontier refers 
         to the highest returns at a given level of risk. A portfolio is said to be efficient 
         if no other portfolio offers higher returns for a lower or equal amount of risk. 
         Where portfolios are located on the efficiency frontier depends on the investor’s 
         degree of risk tolerance.
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const concentrationAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         <b>Concentration</b> Analysis helps you understand the level of exposure to the companies in which 
         you are investing more money. The graph shows the accumulated exposure to top positions, this means 
         if you select 10 stocks in the x axis the y axis shows the percentage of the money invested in 
         the top 10 positions
            </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   const top10CompaniesAlert = () => {
      Swal.fire({
         html: `<div class="align-left">
         Here are the <b>top 10 companies</b> with higher level of investment (Money invested)
         </div>`,
         confirmButtonColor: '#0baca9',
         confirmButtonText: '<a href="about:blank" target="_blank">Learn more</a>',
         showCloseButton: true,
      })
   }

   return (
      <div className="summary_content_wrapper">
         <h1>Risk and Return </h1>

         <div className="RiskReturnGrid">
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Value at risk
                     <button className="moreinfo ml-2" onClick={valueAtRiskAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>

               <div className="">
                  <ValueAtRisk valueatrisk={portfolio_analysis?.var} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Risk vs. return
                     <button className="moreinfo ml-2" onClick={riskVsReturnAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="">
                  <RiskReturns risks={portfolio_analysis?.risk_returns} />
               </div>
            </div>
            <div className="box_chart_area">
               <div className="top_area">
                  <h1>
                     Concentration
                     <button className="moreinfo ml-2" onClick={concentrationAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div>
                  <Concentration
                     concentration={portfolio_analysis?.concentration as never}
                  />
               </div>
            </div>
            <div className="box_chart_area RiskReturnTableAreaWrapper">
               <div className="top_area">
                  <h1>
                     Top 10 companies
                     <button className="moreinfo ml-2" onClick={top10CompaniesAlert}>
                        <Icon.Info />
                     </button>
                  </h1>
               </div>
               <div className="chartRiskReturn" style={{ alignItems: 'flex-start' }}>
                  <table className="RiskReturnTableArea">
                     <thead>
                        <th>Ticker</th>
                        <th>Name</th>
                        <th>% of total</th>
                        <th>Investment</th>
                     </thead>
                     <tbody>
                        {portfolio_analysis?.top_holdings.ticker.map(
                           (ticker: string, key: number) => (
                              <tr key={key}>
                                 <td>{ticker}</td>
                                 <td></td>
                                 <td>
                                    {(
                                       portfolio_analysis.top_holdings.weight[key] *
                                       100
                                    ).toFixed(1)}
                                    %
                                 </td>
                                 <td>
                                    {formatToMoney(
                                       portfolio_analysis.top_holdings
                                          .outstanding_balance[key]
                                    )}
                                 </td>
                              </tr>
                           )
                        )}
                     </tbody>
                  </table>
               </div>
            </div>
         </div>
      </div>
   )
}
