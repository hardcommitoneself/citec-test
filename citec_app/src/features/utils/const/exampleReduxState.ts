// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const exampleReduxState = {
   errors: '',
   taxHarvesting: {
      arrayPositionSelected: [
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      positionsTable: [
         {
            outstanding_balance: 50000,
            ticker: 'AAPL',
            unrealized_capital_gains: -1000,
            taxAlpha: 400,
         },
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      taxAlpha: -400,
      unrealizedCapitalGains: -500,
      unrealizedCapitalGainsNegative: -1000,
      unrealizedCapitalGainsSelected: 0,
   },
   portfolio: {
      ticker: ['$CASH', 'AAPL'],
      outstanding_balance: [1000000, 50000],
      unrealized_capital_gains: [0, 11234],
      analysis: {
         fundamentals: {
            total_assets: 2,
            total_positions: 2,
            pe: 24.32,
            pb: 38.96,
            dividend_yield: 0.59,
         },
         risk: {
            sharpe: 0.305683,
            volatility: 0.0147,
            beta: 0.048719,
            alpha: 0.000035,
         },
         returns: {
            returns_1M: -0.0021,
            returns_3M: -0.0037,
            returns_6M: -0.0028,
            CAGR: -0.1058,
         },
         capital_gains: {
            outstanding_balance: 1050000,
            unrealized_cap_losses: 0,
            unrealized_cap_gains: 11234,
            realized_cap_losses: 0,
            realized_cap_gains: 0,
            taxes_cost: 0,
            taxes_rate: 0.4,
         },
         sectors: {
            sector: ['Cash', 'Information Technology'],
            percentage: [0.952381, 0.047619],
            percentage_benchmark: [0.3333333333333333, 0.6666666666666666],
         },
         factors: {
            factor: ['style', 'size'],
            percentage: [0.502991, 0.52381],
         },
         segment_matrix: {
            style: ['Val', 'Val'],
            size: ['Large', 'Mid'],
            weight: [0, 0],
         },
         top_holdings: {
            ticker: ['$CASH', 'AAPL'],
            weight: [0.952381, 0.047619],
            outstanding_balance: [1000000, 50000],
            unrealized_capital_gains: [0, 11234],
         },
         concentration: {
            ticker: ['$CASH', 'AAPL'],
            concentration: [95.238095, 100],
         },
         var: {
            quantile: [2, 4],
            value: [0.000072, 0.001334],
         },
         risk_returns: {
            ticker: ['portfolio', 'benchmark'],
            returns: [0.004722, 0.119508],
            volatility: [0.008561, 0.205196],
            sharpe: [0.551568, 0.582407],
         },
      },
   },
   universe: {
      factors_filter: [
         {
            factor: 'style',
            ranges: [0, 100],
         },
         {
            factor: 'size',
            ranges: [0, 100],
         },
         {
            factor: 'yield',
            ranges: [0, 100],
         },
      ],
      sectors_filter: [
         'Industrials',
         'Financials',
         'Health Care',
         'Consumer Staples',
         'Communication Services',
         'Information Technology',
         'Consumer Discretionary',
         'Materials',
         'Cash',
         'Real Estate',
         'Energy',
         'Utilities',
      ],
      to_exclude: [],
      to_include: [],
      universe_base: 'SPDR S&P 500',
   },
   rebalance: {
      ticker: ['$CASH', 'AAPL'],
      hold: [0, 1],
      objective: {
         name: 'Minimize tracking error',
         key: 'min_tracking_error',
      },
      benchmark: 'SPY',
      min_weight: 0.000001,
      number_stocks: [10, 200],
   },
   portfolio_optimized: {
      portfolio_id: null,
      portfolio: 'optimized_portfolio',
      benchmark: 'SPY',
      creation_date: null,
      ticker: ['AAPL', 'MSFT'],
      outstanding_balance: [72506.52, 63477.37],
      unrealized_capital_gains: [0, 0],
      optimization_status: 'Optimization terminated successfully',
   },
   trades: [
      {
         ticker: 'MRK',
         operation: 'SELL',
         volume: 275161.76,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 3053,
         expected_volume: 275193,
         delta_volume: 31,
         price: 90.1388,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/2eedcaa6-80ec-11ea-853f-00000000092a.png',
            name: 'Merck & Co Inc',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'BMY',
         operation: 'SELL',
         volume: 152166.26,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 2072,
         expected_volume: 152146,
         delta_volume: -19,
         price: 73.43,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/8b628318-80eb-11ea-93a2-00000000092a.png',
            name: 'Bristol-Myers Squibb Co',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'WMT',
         operation: 'SELL',
         volume: 83898.27,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 679,
         expected_volume: 83845,
         delta_volume: -53,
         price: 123.4832,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/52057505dd05a752d364065fcddf7772a700b31c47c39e9dcb41e393f6693d52.png',
            name: 'Walmart Inc',
         },
         gain_loss: [0, 0],
      },
   ],
   workflow: {
      type: 'load',
   },
   static_data: {
      universe: {
         universes: [
            {
               name: 'INVESCO QQQ TRUST SERIES 1',
               description: 'INVESCO QQQ TRUST SERIES 1',
            },
            {
               name: 'ISHARES MSCI USA ESG SELECT',
               description: 'ISHARES MSCI USA ESG SELECT',
            },
         ],
         sectors: [
            'Cash',
            'Communication Services',
            'Consumer Discretionary',
            'Consumer Staples',
         ],
         factors: ['style', 'size', 'yield', 'quality'],
      },
      rebalance: {
         objectives: [
            {
               key: 'min_tracking_error',
               name: 'Minimize tracking error',
            },
            {
               key: 'min_risk_parity',
               name: 'Minimize risk parity',
            },
            { key: 'min_volatility', name: 'Minimize volatility' },
            {
               key: 'max_diversification_ratio',
               name: 'Maximize diversification ratio',
            },
         ],
         benchmarks: [
            {
               key: 'sp_500',
               description: 'S&P 500 universe',
               ticker: 'SPY',
            },
            {
               key: 'nasdaq_100',
               description: 'NASDAQ 100',
               ticker: 'QQQ',
            },
         ],
      },
   },
   brokers: {
      brokerConnected: '',
      balance: 0,
      accountType: [],
      a1c_t: '',
   },
   ideas: [
      {
         portfolio_id: 16,
         portfolio: 'my_first_portfolio_sadasd',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 18,
         portfolio: 'testing',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 19,
         portfolio: 'Jueves',
         benchmark: 'SPY',
         creation_date: '2022-06-23 10:38:00',
         ticker: [
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'AAPL',
            'MMM',
            'TMO',
            'BLK',
            'MSFT',
            'QCOM',
            'NVDA',
            'CSX',
            'ADBE',
            'XOM',
            'AVGO',
            'BAC',
            'TSLA',
            'SHW',
            'TXN',
            'NFLX',
            'GOOGL',
            'C',
            'MMC',
            'BKNG',
            'COST',
            'NKE',
            'DIS',
            'EL',
            'CSCO',
            'BRK.B',
            'META',
            'AMZN',
            'DG',
            'CCI',
            'ORCL',
            'USB',
            'TGT',
            'PG',
            'ICE',
            'GOOG',
            'AMD',
            'MCD',
            'CVS',
            'HD',
            'CME',
            'MU',
            'UNH',
            'ATVI',
            'V',
            'EOG',
            'GS',
            'HUM',
            'COP',
         ],
         outstanding_balance: [
            80352.37, 74072.31, 59343.2, 54736.44, 41924.05, 41510.68, 35441.43,
            27349.74, 26308.48, 26136.99, 25646.73, 25479.44, 24726.53, 24075.27,
            23856.18, 23763.38, 23735.12, 23092.02, 22995.95, 21888.81, 21775.5,
            21090.31, 16734.94, 16192.87, 15720.33, 14020.04, 14018.61, 13225.5,
            12996.74, 11693.52, 11315.64, 10798.53, 10583.72, 10309.54, 10245.54,
            9611.94, 8747.21, 8723.18, 6339.86, 5866.02, 5426.94, 4912.02, 4857.23,
            3915.03, 3395.13, 3370.41, 3322.34, 3184.94, 3001.78, 1983.75, 1858.2,
            1789.12, 1309.98, 1228.49,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 20,
         portfolio: 'Min volatility',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:25:41',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 21,
         portfolio: 'min2',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:27:32',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
   ],
}
