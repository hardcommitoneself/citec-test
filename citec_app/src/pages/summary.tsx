// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { Button, Tooltip } from '@mui/material'
import { useState } from 'react'
import Swal from 'sweetalert2'

import { AppNavigate } from 'routes'
import { Loading } from 'features/ui'
import { PageContainer } from 'features/ui/page-container'
import {
   useAppNavigate,
   useGetSummary,
   useValidationCriteriaSelected,
} from 'features/utils/hooks'
import { useAppSelector } from 'store/hooks'
import { Backtest, Overview, Risk, Trades } from 'features/summary/tabs'
import { savePortfolio } from 'features/utils/save-portfolio'
import { renderPopup } from 'features/popups/backtest-popups'

type ActiveViews = 'trades' | 'overview' | 'risk' | 'backtest'

const Group = require('assets/img/group.png')

export const Summary = () => {
   const { isLoading, backtest, analysis } = useGetSummary()

   const error = useAppSelector((state) => state.errors)
   const { ticker, outstanding_balance, unrealized_capital_gains } = useAppSelector(
      (state) => state.portfolio_optimized
   )
   const benchmark = useAppSelector((state) => state.rebalance.benchmark)

   const [activeView, setActiveView] = useState<ActiveViews>('trades')

   const navigate = useAppNavigate()
   const views = ['Trades', 'Overview', 'Risk', 'Backtest']
   const disabledViews = ['']

   const isCriteriaSelected = useValidationCriteriaSelected()
   if (!isCriteriaSelected) {
      Swal.fire({
         icon: 'error',
         title: 'Oops...',
         text: 'You need to upload select a benchmark and objective',
      })

      return <AppNavigate to="/criteria" />
   }

   const handleSavePortfolio = () => {
      savePortfolio({
         navigate,
         benchmark,
         outstanding_balance,
         ticker,
         unrealized_capital_gains,
      })
   }

   if (isLoading) return <Loading />

   return (
      <PageContainer error={error}>
         <div className="dashboard_wrapper">
            <div className="InvestmentBody summary dashboard_body">
               <div className="inner_dashboard_cri mt-0">
                  <h1>
                     <span>Congrats!</span> You have created a portfolio. Check if
                     you like it
                  </h1>

                  <div className="options_summary_buttons">
                     {views.map((view) => (
                        <button
                           key={view}
                           className={
                              activeView === view.toLowerCase()
                                 ? 'active-btn'
                                 : 'null'
                           }
                           disabled={disabledViews.includes(view)}
                           onClick={() => {
                              setActiveView(view.toLowerCase() as ActiveViews)
                           }}
                           data-testid={`btn-${view.toLowerCase}`}
                        >
                           {view}
                        </button>
                     ))}
                  </div>
                  <div className="bg-white p-2" data-testid={`active-${activeView}`}>
                     <div className="summary_content_wrapper">
                        { activeView !== 'trades' && (
                           <div className="action_summary">
                              {activeView === 'backtest' && (
                                 <>
                                    <h1>Performance Backtest</h1>
                                    <img
                                       className="moreinfo"
                                       src={Group}
                                       alt=""
                                       onClick={() => {
                                          renderPopup('perfBacktestAlert')
                                       }}
                                       style={{ marginLeft: '15px' }}
                                       data-testid="info-summary-backtest"
                                    />
                                 </>
                              )}
                              {activeView === 'overview' && <h1>Overview</h1>}
                              {activeView === 'risk' && <h1>Risk and Return</h1>}
                              <div className='button_download_trades'>
                                 <Button
                                    className="h-[35px] min-w-0 rounded"
                                    variant="save"
                                    onClick={handleSavePortfolio}
                                 >
                                    <span>Save</span>
                                 </Button>
                              </div>
                           </div>
                        )}
                        {activeView === 'trades' && <Trades />}
                        {activeView === 'backtest' && !!backtest && (
                           <Backtest {...backtest} />
                        )}
                        {activeView == 'overview' && (
                           <Overview backtest={backtest} analysis={analysis} />
                        )}
                        {activeView == 'risk' && <Risk analysis={analysis} />}
                     </div>
                  </div>

               </div>
            </div>
         </div>
      </PageContainer>
   )
}
