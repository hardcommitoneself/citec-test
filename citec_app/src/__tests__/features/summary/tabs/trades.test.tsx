// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

import { mount, shallow } from 'enzyme'
import { Trades } from 'features/summary/tabs'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'

describe('Test in Trades tab', () => {
   const mockStore = configureStore()
   const store = mockStore(exampleReduxState)

   const wrapper = shallow(
      <Provider store={store}>
         <Trades />
      </Provider>
   )

   test('snapshot', () => {
      expect(wrapper).toMatchSnapshot()
   })

   it('click on button download icons', () => {
      const component = mount(
         <Provider store={store}>
            <BrowserRouter>
               <Trades />
            </BrowserRouter>
         </Provider>
      )

      component
         .find({ 'data-testid': 'info-summary-trades-download' })
         .simulate('click')
   })
})
