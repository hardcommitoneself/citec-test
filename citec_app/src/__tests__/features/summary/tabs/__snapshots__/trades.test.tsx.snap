// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Trades tab snapshot 1`] = `ShallowWrapper {}`;
